custom_components/waste_collection_schedule/waste_collection_schedule/service/EcoHarmonogramPL.py

```diff
@@ -0,0 +1,66 @@
+import sys
+
+import requests
+
+towns_url = "https://ecoharmonogram.pl/api/api.php?action=getTowns"
+scheduled_periods_url = "https://ecoharmonogram.pl/api/api.php?action=getSchedulePeriods"
+streets_url = "https://ecoharmonogram.pl/api/api.php?action=getStreets"
+schedules_url = "https://ecoharmonogram.pl/api/api.php?action=getSchedules"
+
+headers = {
+    'Content-Type': 'application/json; charset=utf-8',
+    'Accept': 'application/json',
+}
+
+
+class Ecoharmonogram:
+    @staticmethod
+    def fetch_schedules(sp, street):
+        schedules_response = requests.get(
+            schedules_url + "&streetId=" + street.get("id") + "&schedulePeriodId=" + sp.get("id"),
+            headers=headers)
+        schedules_response.encoding = "utf-8-sig"
+        schedules_response = schedules_response.json()
+        return schedules_response
+
+    @staticmethod
+    def fetch_streets(sp, town, street, house_number):
+        streets_response = requests.get(
```

Comment:
Please pass parameters in URLs using `requests` `param` argument:

https://requests.readthedocs.io/en/latest/user/quickstart/#passing-parameters-in-urls

Reply:
@mampfes done - nice spot